### PR TITLE
Jn statsdv2

### PR DESCRIPTION
**Pull request by cypriss**

### CODE REVIEW

File: statsd_sink.go [hunk 1]
```diff
@@ -2,164 +2,413 @@ package health
...
+		return nil, err
+	}
+
+	s := &StatsDSink{
 		SanitizationFunc: sanitizeKey,
-		conn:             c,
+		udpConn:          c.(*net.UDPConn),
+		udpAddr:          ra,
+		cmdChan:          make(chan statsdEmitCmd, cmdChanBuffSize),
+		drainDoneChan:    make(chan int),
```
Comment (taylorchu) [inline statsd_sink.go]: chan struct{} is enough because we do not use int value from channel.

File: statsd_sink.go [hunk 2]
```diff
@@ -2,164 +2,413 @@ package health
...
+	Status CompletionStatus
 }
 
-func NewStatsDSink(addr, prefix string) (Sink, error) {
-	c, err := net.DialTimeout("udp", addr, 2*time.Second)
+const cmdChanBuffSize = 8192 // random-ass-guess
+const maxUdpBytes = 1440     // 1500(Ethernet MTU) - 60(Max UDP header size
+
+func NewStatsDSink(addr, prefix string) (*StatsDSink, error) {
+	c, err := net.ListenPacket("udp", ":0")
```
Comment (taylorchu) [inline statsd_sink.go]: I don't understand why we need to listen.

File: statsd_sink.go [hunk 3]
```diff
@@ -2,164 +2,413 @@ package health
...
+	eventBytes := b.Bytes()
+
+	return eventBytes, jobEventBytes
 }
 
-func shouldSanitize(r rune) bool {
-	switch r {
-	case '|', ':':
-		return true
+func (s *StatsDSink) gaugeBytes(job, event string, value float64) ([]byte, []byte) {
```
Comment (taylorchu) [inline statsd_sink.go]: timing?

Comment (cypriss) [inline statsd_sink.go]: you mean what's the benchmark for the new sanitize?

**Pull request closed without merging**